### PR TITLE
feat: update data loader

### PR DESCRIPTION
**Pull request by rkarmaka**

### CODE REVIEW

File: src/napari_sam_prompt/_sam_prompt_widget.py [hunk 1]
```diff
@@ -265,12 +265,40 @@ def _convert_image(self, layer_name: str) -> np.ndarray:
...
+        if channels == 1:
+            return np.stack([data] * 3, axis=-1)
+        elif channels == 2:
+            avg_channels = np.mean(data, axis=-1, keepdims=True)
+            return np.concatenate([data, avg_channels], axis=-1)
+        elif channels == 3:
+            return data
+        else:
+            avg_channels = np.mean(data, axis=-1, keepdims=True)
+            return np.stack([avg_channels] * 3, axis=-1)
```
Comment (fdrgsp) [inline src/napari_sam_prompt/_sam_prompt_widget.py:301]:
```suggestion
    def _convert_8bit(self, data: np.ndarray) -> np.ndarray:
        """Convert image to 8-bit."""
        for i in range(3):
            temp_data = data[:, :, i]
            data[:, :, i] = (
                255
                * ((temp_data - temp_data.min()) / (temp_data.max() - temp_data.min()))
            ).astype("uint8")
        return data

    def _convert_to_three_channels(self, data: np.ndarray) -> np.ndarray:
        """Convert the image to 3 channels."""
        channels = data.shape[-1] if len(data.shape) == 3 else 1
        if channels == 1:
            return np.stack([data] * 3, axis=-1)
        elif channels == 2:
            avg_channels = np.mean(data, axis=-1, keepdims=True)
            return np.concatenate([data, avg_channels], axis=-1)
        elif channels == 3:
            return data
        else:
            avg_channels = np.mean(data, axis=-1, keepdims=True)
            return np.stack([avg_channels] * 3, axis=-1)
```

Comment (fdrgsp) [reply src/napari_sam_prompt/_sam_prompt_widget.py:301]: Just adding typing to the functions.

Comment (fdrgsp) [reply src/napari_sam_prompt/_sam_prompt_widget.py:301]: also I opened an image with shape (2, 1128, 1024) and I got the IndexError:

```
 data_8bit = self._convert_8bit(data_three_channels)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fdrgsp/Documents/git/napari-sam-prompt/src/napari_sam_prompt/_sam_prompt_widget.py", line 277, in _convert_8bit
    temp_data = data[:, :, i]
                ~~~~^^^^^^^^^
IndexError: index 1 is out of bounds for axis 2 with size 1
```

Can you add support for this images too? Thanks!

**Review by fdrgsp — CHANGES_REQUESTED** [review]

File: src/napari_sam_prompt/_sam_prompt_widget.py [hunk 2]
```diff
@@ -281,12 +281,36 @@ def _convert_image(self, layer_name: str) -> np.ndarray:
...
+            return np.stack([data] * 3, axis=-1)
+        elif channels == 2:
+            avg_channels = np.mean(data, axis=-1, keepdims=True)
+            return np.concatenate([data, avg_channels], axis=-1)
+        elif channels == 3:
+            return data
+        else:
+            avg_channels = np.mean(data, axis=-1)
+            return np.stack([avg_channels] * 3, axis=-1)
+            return np.stack([avg_channels] * 3, axis=-1)
```
Comment (fdrgsp) [inline src/napari_sam_prompt/_sam_prompt_widget.py:313]:
```suggestion

```

**Review by fdrgsp — APPROVED** [review]

**Review by rkarmaka — COMMENTED** [review]
looks good

**Pull request closed without merging**